### ISSUE TITLE
Bricked Bios with update

### ISSUE THREAD

**Issue by Huberer01** [zearp/OptiHack#37]
Hello,
unfortunately something went wrong with the bios update of the modifications I did. Now my system is not booting anymore. Would you be so kind and post pictures of the bios chip on the mobo and how the cable of the programmer is connected to the programmer it selves (24xxx or 25xxx eeproms)
Thanks in advance

Edit: I have a Dell 9020 MT, just for information

**Comment by zearp** [comment]
You will need a SOIC8/SOP8 programmer kit to flash back backups (.bin files) to the chips. I bricked mine a few times, the hardest part is fitting the clip on the tiny chips. I used a Raspberry Pi with Debian and flashrom but any laptop or computer will do.

Don't use flashrom for Windows or any other Windows tools as they use very crappy drivers which will give you issues when reading/writing chips. Stick to flashrom in your fav Linux distro.

The board has 2 chips, I'm unable to provide pictures (also I only have 7020s) but they are labeled on the board itself, you can't really miss them.

You will need to flash back the 4mb and 8mb bin files you should have if you followed my guide. I added nearly complete [recovery instructions](https://github.com/zearp/OptiHack/blob/master/text/BIOS_STUFF.md#recovery) to it, please check it out.

In short what you need to do is connect to each chip, scan the bus with flashrom and flash back the matching backup; 4mb file to 4mb chip and 8mb file to 8mb chip. The chip name will contain an indication to the size, 32 would indicate the 4mb chip and 64 the 8mb chip.

I bought [this](http://keeyees.com/a/Products/ej/36.html) kit from Amazon cuz I couldn't wait for post from China. Amazon.de [has](https://www.amazon.de/-/en/gp/product/B07SNTL5V6) it in stock themselves with Prime. You can find it on other Amazon websites by searching for ```B07SNTL5V6```. 

Good luck!

**Comment by Huberer01** [comment]
Thanks, I already have the equipment, but I can't find the chips on the mobo. Are they closed to the dimm sockets? I found two with the naming 25Q064A and 25Q03213 but I can't find any hints on the net that these are the bios chips.
Closed to them there's the naming SPI_2 and SPI_3. SPI stands for the bios chips but I can't find SPI_1 
Can you remember where these chips are located approximately?

**Comment by zearp** [comment]
There are only 2 BIOS chips on there, I'm not sure how mine were labeled but SPI sounds about right. No clue why you have 2 and 3 but if one of them is 4mb and the other is 8mb it can't be anything else.

As for the location I don't have any 9020s and their board is different from the 7020. They just share the chipset but the board itself is different.

If I were you I would search on the chip names found near the SPI labels if you're unsure which is which. They are small 8 pin chips. I think mine had Winbond printed on them. It's been a while since I had to open one of these.

![Screenshot 2021-02-27 at 13 05 50](https://user-images.githubusercontent.com/63272687/109386672-8990ec00-78fc-11eb-851b-55fc491423f2.png)

**Comment by Huberer01** [comment]
The above mentioned chips were the bios chips. Unfortunately my backup-bios got corrupted and so I couldn't flash it anymore. Fortunately I found two working bin's on the www and with them I could get my system working again.
I've a second Dell 9020 in the office and I will clone this bios to the "broken" machine and with a flashing tool I will change the mac-address of the lan-card. With this I'm optimistic that my "broken" machine will work as usual.
Thanks for your help. This issue is now solved.
